lib/repository/club_member/club_member_term_repository.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:dio/dio.dart';
+import 'package:woohakdong/model/club_member/club_member_term.dart';
+
+import '../../service/dio/dio_service.dart';
+import '../../service/logger/logger.dart';
+
+class ClubMemberTermRepository {
+  final Dio _dio = DioService().dio;
+
+  Future<List<ClubMemberTerm>> getClubMemberTermList(int clubId) async {
+    try {
+      logger.i('동아리 가입 기수 리스트 조회 시도');
+
+      final response = await _dio.get('/clubs/$clubId/terms');
+
+      if (response.statusCode == 200) {
+        final Map<String, dynamic> jsonData = response.data;
+
+        List<dynamic> clubMemberTermListData = jsonData['result'] as List<dynamic>;
+
+        return clubMemberTermListData.map((json) => ClubMemberTerm.fromJson(json as Map<String, dynamic>)).toList();
+
+      }
+
+      throw Exception();
```

Comment:
혹시 early return하는 방식은 어떨까요? [참고](https://dcm.dev/docs/rules/common/prefer-early-return/)

지금도 보기엔 어려움은 없지만, 후에 if문에서 여러 라인을 포함할 때 depth를 깊지 않게 가져갈 수 있어 가독성이 낫지 않을까 싶네요!
```suggestion
      if (response.statusCode != 200) {
        throw Exception();
      }
     
      final Map<String, dynamic> jsonData = response.data;

      List<dynamic> clubMemberTermListData = jsonData['result'] as List<dynamic>;

      return clubMemberTermListData.map((json) => ClubMemberTerm.fromJson(json as Map<String, dynamic>)).toList();
```

Reply:
오 저번에 알려 주신 방법으로 else를 없애긴 했는데, 저런 방법도 있군요!